src/ocean_data_assim/MOM_oda_driver.F90

```diff
@@ -438,20 +518,62 @@ subroutine oda(Time, CS)
 
     !! switch to global pelist
     call set_PElist(CS%filter_pelist)
-
+    call cpu_clock_begin(id_clock_oda_filter)
     call get_profiles(Time, CS%Profiles, CS%CProfiles)
 #ifdef ENABLE_ECDA
     call ensemble_filter(CS%Ocean_prior, CS%Ocean_posterior, CS%CProfiles, CS%kdroot, CS%mpp_domain, CS%oda_grid)
 #endif
-
+    call cpu_clock_end(id_clock_oda_filter)
+    !if (CS%write_obs) call save_obs_diff(CS%CProfiles) ! not fully implemented
     !! switch back to ensemble member pelist
     call set_PElist(CS%ensemble_pelist(CS%ensemble_id,:))
+    call get_posterior_tracer(Time, CS, increment=.true.)
+    if (CS%do_bias_adjustment) call get_bias_correction_tracer(Time, CS)
 
   endif
 
   return
 end subroutine oda
 
+subroutine get_bias_correction_tracer(Time, CS)
+    type(time_type), intent(in) :: Time !< the current model time
+    type(ODA_CS), pointer :: CS !< ocean DA control structure
+
+    integer :: i,j,k
+    real, allocatable, dimension(:,:,:) :: T_bias, S_bias
+    real, allocatable, dimension(:,:,:) :: mask_z
+    real, allocatable, dimension(:), target :: z_in, z_edges_in
+    real :: missing_value
+    integer,dimension(3) :: fld_sz
+
+    call cpu_clock_begin(id_clock_bias_adjustment)
+    call horiz_interp_and_extrap_tracer(CS%INC_CS%T_id,Time,1.0,CS%G,T_bias,&
+            mask_z,z_in,z_edges_in,missing_value,.true.,.false.,.false.,.true.)
+    call horiz_interp_and_extrap_tracer(CS%INC_CS%S_id,Time,1.0,CS%G,S_bias,&
+            mask_z,z_in,z_edges_in,missing_value,.true.,.false.,.false.,.true.)
+
+    ! This should be replaced to use mask_z instead of the following lines
+    ! which are intended to zero land values using an arbitrary limit.
+    fld_sz=shape(T_bias)
+    do i=1,fld_sz(1)
+       do j=1,fld_sz(2)
+          do k=1,fld_sz(3)
+             if (T_bias(i,j,k) .gt. 1.0E-3) T_bias(i,j,k) = 0.0
+             if (S_bias(i,j,k) .gt. 1.0E-3) S_bias(i,j,k) = 0.0
```

Comment:
Do the 1e-3 terms need dimensional scaling?

Reply:
I don't think the tracer values are subject to dimensional scaling. There is a comment that this should be replaced to use mask_z instead, which I would concur with. @feiyulu , would switching to use a logical mask impact your answers?